common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -77,47 +79,21 @@
     public MsalOAuth2TokenCache(final Context context,
                                 final IAccountCredentialCache accountCredentialCache,
                                 final IAccountCredentialAdapter<
-                                        MicrosoftStsOAuth2Strategy,
-                                        MicrosoftStsAuthorizationRequest,
-                                        MicrosoftStsTokenResponse,
-                                        MicrosoftAccount,
-                                        MicrosoftRefreshToken> accountCredentialAdapter) {
+                                        GenericOAuth2Strategy,
+                                        GenericAuthorizationRequest,
+                                        GenericTokenResponse,
+                                        GenericAccount,
+                                        GenericRefreshToken> accountCredentialAdapter) {
         super(context);
         Logger.verbose(TAG, "Init: " + TAG);
         mAccountCredentialCache = accountCredentialCache;
-        mSharedSsoCaches = new ArrayList<>();
-        mAccountCredentialAdapter = accountCredentialAdapter;
-    }
-
-    /**
-     * Constructor of MsalOAuth2TokenCache.
-     *
-     * @param context                  Context
-     * @param accountCredentialCache   IAccountCredentialCache
-     * @param accountCredentialAdapter IAccountCredentialAdapter
-     * @param sharedSsoCaches          List<IShareSingleSignOnState>
-     */
-    public MsalOAuth2TokenCache(final Context context,
-                                final IAccountCredentialCache accountCredentialCache,
-                                final IAccountCredentialAdapter<
-                                        MicrosoftStsOAuth2Strategy,
-                                        MicrosoftStsAuthorizationRequest,
-                                        MicrosoftStsTokenResponse,
-                                        MicrosoftAccount,
-                                        MicrosoftRefreshToken> accountCredentialAdapter,
-                                final List<IShareSingleSignOnState> sharedSsoCaches) {
-        super(context);
-        Logger.verbose(TAG, "Init: " + TAG);
-        mAccountCredentialCache = accountCredentialCache;
-        mSharedSsoCaches = sharedSsoCaches;
         mAccountCredentialAdapter = accountCredentialAdapter;
     }
 
     @Override
-    public ICacheRecord save(
-            final MicrosoftStsOAuth2Strategy oAuth2Strategy,
-            final MicrosoftStsAuthorizationRequest request,
-            final MicrosoftStsTokenResponse response) throws ClientException {
+    public ICacheRecord save(@NonNull final GenericOAuth2Strategy oAuth2Strategy,
```

Comment:
Save whatever you want -- the adapter's job is to make this work (which you supply when constructing the cache)